### DISCUSSION TITLE
Redux object into "reset" function

### DISCUSSION THREAD

**Discussion by Flo0806** [react-hook-form/react-hook-form#8060] [Q&A] [answered]
Hello!

In my redux store i save data from a http request. The data looks like this:
` {
        name: "Recipe",
        ingredients: [
          { name: "Ingredient 1", amount: "1tl" },
          { name: "Ingredient 2", amount: "2tl" },
        ],
}`

So... if i set the response directly with the reset function `reset(response.data)` (for example) then the ingredients array (which is used by useFieldArray) works perfectly! But if i set the response into a redux Object (`state.currentRecipe = action.payload`) and then i set the currentRecipe with the reset function (`reset({...currentRecipe})` or `reset(currentRecipe)` as example) then the follow error throw:
`Uncaught TypeError: Cannot delete property 'name' of #<Object>
    at unset (unset.ts:26:1)
    at Object.unregister (createFormControl.ts:909:1)
    at useController.ts:91:1
...........`

In the unset.ts the property cannot deleted.... If i do the follow workaround (isn't it?) `reset(JSON.parse(JSON.stringify(currentRecipe)))` all works fine without error.

So my question: What does redux do wrong or what I doing wrong?

Note: I use defaultValues with useForm for the ingredients too like:
`defaultValues: {
    name: "New Recipe",
    ingredients: [{name: "Empty Ingredient", amount: "1"}]
}`

Greetings, Florian

**Comment by Moshyfawn** [comment, marked as answer]
Greetings, @Flo0806!

You might wanna provide a minimal repro codesanbox to help us understand your use-case.

_I quit using Redux quite some time ago, so I might be missing some new state management quirk :P_

**Reply by Flo0806** [reply]
Yes, i can and i have. https://codesandbox.io/s/autumn-star-s5p1b3?file=/src/App.js
So start the box and you see the error.

Thx, Florian

**Reply by Moshyfawn** [reply]
Thx for the the clear repro!

So, what I found is that if you set the `shouldUnregister: true` flag at the form level _(pass it to the `useForm` hook)_, it works as expected. IDK, why setting the flag at the field level would throw when it's the same logic, I believe.

Might be related #932.
@bluebill1049, as you've possibly already experienced the issue, can you advise?

**Reply by bluebill1049** [reply]
That's a similar issue with custom objects: https://github.com/react-hook-form/react-hook-form/issues/7279
workaround: https://codesandbox.io/s/modest-wave-dn0is6?file=/src/App.js

**Reply by Moshyfawn** [reply]
Doesn't RTK return a plain JS object? IK it uses Immer, but the result is a simple object, from what I've seen

**Reply by Flo0806** [reply]
Hello again!

Thx for this fast support :-) Wow! So i have searched a little between "custom objects" and the error itself. And i have found the follow: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Non_configurable_array_element 
Here you can see how to reproduce a similar error when using the `Object.defineProperty` function. Then i have looked into ImmerJS and i have see the lib works many with the `Object.defineProperty` function.
And when you debug unset.js in the browser, you see the error occurs when ChildObject looks like: `amount: {}`. But (in my case) amount has a string value not an Object. 
I don't know..... But eventually it's the problem.

And i have updated my Sandbox example with react-hookstore (a store made with react hooks). And it works without errors. 

Thx for the workaround @bluebill1049 

Greetings Flo

**Reply by derek-primumco** [reply]
> shouldUnregister: true

This worked for me. Thanks!

> if you set the `shouldUnregister: true` flag at the form level _(pass it to the `useForm` hook)_, it works as expected

This worked for me. Thanks!
